worldguard-core/src/main/java/com/sk89q/worldguard/protection/managers/migration/WorldHeightMigration.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldguard.protection.managers.migration;
+
+import com.sk89q.worldedit.world.World;
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.protection.flags.registry.FlagRegistry;
+import com.sk89q.worldguard.protection.managers.storage.RegionDatabase;
+import com.sk89q.worldguard.protection.managers.storage.RegionDriver;
+import com.sk89q.worldguard.protection.managers.storage.StorageException;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Field;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WorldHeightMigration extends AbstractMigration {
+    private static final Logger log = Logger.getLogger(WorldHeightMigration.class.getCanonicalName());
+
+    private static final Field minField;
+    private static final Field maxField;
+    static {
+        try {
+            minField = ProtectedRegion.class.getDeclaredField("min");
+            minField.setAccessible(true);
+            maxField = ProtectedRegion.class.getDeclaredField("max");
+            maxField.setAccessible(true);
+        } catch (NoSuchFieldException e) {
+            throw new ExceptionInInitializerError(new MigrationException("Migrator broke.", e));
+        }
+    }
+
+    private final FlagRegistry flagRegistry;
+    private final World world;
+
+    public WorldHeightMigration(RegionDriver driver, FlagRegistry flagRegistry, @Nullable World world) {
+        super(driver);
+        this.flagRegistry = flagRegistry;
+        this.world = world;
+    }
+
+    @Override
+    protected void migrate(RegionDatabase store) throws MigrationException {
+        if (world != null && !store.getName().equals(world.getName())) return;
+
+        log.log(Level.INFO, "Migrating regions in '" + store.getName() + "' to new height limits...");
+
+        Set<ProtectedRegion> regions;
+
+        try {
+            regions = store.loadAll(flagRegistry);
+        } catch (StorageException e) {
+            throw new MigrationException("Failed to load region data for the world '" + store.getName() + "'", e);
+        }
+
+        int min = -64;
+        int max = 319;
+        World world = WorldGuard.getInstance().getPlatform().getMatcher().getWorldByName(store.getName());
+        if (world != null) {
+            min = world.getMinY();
+            max = world.getMaxY();
+        }
+        for (ProtectedRegion region : regions) {
+            if (region.getMinimumPoint().getBlockY() == 0
```

Comment:
Maybe change it from min=0 and max=255 to min<0 and max>255 for people with datapacks in 1.17.
Or stupid developers (such as me) who created regions from 0...256